src/components/DisplayProfile.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect } from "react";
+import { getUserData } from "../modules/getUserData";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom"
+
+const DisplayProfile = props => {
+  const userDataGrab = async () => {
+    if (props.userAttrs) {
+      props.changeUserShowData(await getUserData(props.userAttrs.id));
+    }
+  };
+
+  useEffect(() => {
+    userDataGrab();
+  }, [props.userAttrs]);
+
+  return (
+    <div id="profile">
+      {props.userShowData ? (
+        props.userShowData.email ? (
+          <>
```

Comment:
what is the difference between the first two conditions here?

Reply:
First makes sure that userShowData exists, if not then it renders please log in. Second one checks if it has properly been loaded with the showData and not only the error message.

Reply:
okay cool